engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -15,16 +15,15 @@ def self.update_cache(cache_entry, mode, registry: false)
     )
   end
 
-  def self.build_cache_entry(remote_ip, system_login, encoded_reg_code, mode, product)
+  def self.build_cache_entry(remote_ip, system_login, encoded_reg_code, mode, product, product_class) # rubocop:disable Metrics/ParameterLists
     if mode == 'payg'
       [remote_ip, system_login, product.id].join('-')
     elsif mode == 'registry'
       [remote_ip, system_login].join('-')
     else
       # for byos or hybrid cache
-      product_hash = product.attributes.symbolize_keys.slice(:identifier, :version, :arch)
-      product_triplet = "#{product_hash[:identifier]}_#{product_hash[:version]}_#{product_hash[:arch]}"
-      "#{encoded_reg_code}-#{product_triplet}-active"
+      # product is the product_class here
+      "#{encoded_reg_code}-#{product_class}-active"
```

Comment:
I thought we were adding the system identifier to the cache entry, instanceid for EC2, vmid for Azure machineid for GCE?

Reply:
that is done in a different PR

This PR adds a subscription cache check